### ISSUE TITLE
Xgboost output considered a probability

### ISSUE THREAD

**Issue by naxokite** [dmlc/xgboost#2312]
Hi all, I would like to understand why the output of the model (for binary classification) is broadly considered to be a probability (even in the official documentation here in the repo). I understand that the scores are in [0, 1] but I also understand this is not enough to actually consider it a probability.
What, in the training process (objective function, minimization strategy...) enables the user to consider the output of the model the probability p(y | X)?

Just to clarify, I am not asking about the leaf scores of each booster being negative or anything like this. I am looking for the theoretical concept allowing us to go from the logistic transformation of the weights minimizing the obj function (leaf scores) to a probability distribution.

Any help is much appreciated.

Best.

**Comment by khotilov** [comment]
https://en.wikipedia.org/wiki/Logistic_regression
https://en.wikipedia.org/wiki/Calibration_(statistics)
Under-trained boosting models (too few iterations with small learning rate) would typically be under-calibrated.

**Comment by naxokite** [comment]
Posting 2 wikipedia links doesn't solve any issue.
As I understand it there is no logistic regression involved here, the weight leafs go through the logistic function but that does not make the output a probability.
I understand calibration concepts, my question is more in the line of "does xgb implement any of them?"
Please do not close any issues at your own will, let the people who opened them do that when they feel they got a response.

Best.

**Comment by khotilov** [comment]
@naxokite What is a *specific xgboost-related issue* here that needs a solution? I've tried to guess the most likely specific issue, but apparently it wasn't about that. Your question, as it is, is of rather general nature, and would be more appropriate for stackoverflow or CrossValidated. Optimizing logistic loss involves some amount of implicit calibration, and there's not much theoretical difference in this respect to logistic regression. And there are many similarities b/w gradient boosting with logistic loss and logistic regression since GB is essentially a generalized additive model.

**Comment by naxokite** [comment]
@khotilov in the [_xgboost-related documentation_](https://github.com/dmlc/xgboost/tree/master/demo/binary_classification), you can find that "**For binary classification, the output predictions are probability confidence scores in [0,1], corresponds to the probability of the label to be positive.**". This is the sentence I would like to understand more in detail, following the [_xgboost-related documentation_](http://homes.cs.washington.edu/~tqchen/data/pdf/BoostedTree.pdf), the relationship between THIS gradient boosting implementation and the outputs being considered probabilities, as far as I can read in those slides there is no explicit calibration whatsoever.
I don't want to understand the relationship between gradient boosting in general and probas, I am interested in this implementation because it is the one I use and the one where I found that its output could be considered a probability.
It would be great to have the issue reopened so I can get help.

Thank you very much.

**Comment by khotilov** [comment]
@naxokite This documentation statement does not give a definitive guarantee that the outputs would be true probabilities. And there's no explicit calibration in xgboost. Any post-processing you would deem needed is up to you.

And again: the main question you are asking is not specific to this implementation. I've googled some more explanations for you: https://www.quora.com/Why-is-the-output-of-logistic-regression-interpreted-as-a-probability . If you don't want to learn the fundamentals and you don't recognize help when you are given it - that's your issue. We have neither time nor obligation to help with that here.

**Comment by naxokite** [comment]
@khotilov Trying to understand the fundamentals here, I understand your point about interpreting this as a more general question (and you're right) and not this implementation but my question was actually about the sentence found in the documentation of this implementation, that's why I asked it here. Wanted to now is some post-processing was being done in this implementation to actually consider it as a probability.
I appreciate the time and effort you have dedicated to me but that was completely my point, maybe if the issue hadn't been closed right after you posted the first answer (your last one was much more helpful than that) maybe other people could have helped in this discussion, the vigilante attitude towards open issues is also something to avoid.

Thanks again for the help and apologies for any inconvenience.

Nacho.
